### PR TITLE
Update docs lists_mergeby

### LINKED ISSUE #8474: Complete examples and documentation of the filter lists_merge_by
### Summary

The filter lists_mergeby (quoting from  plugins/filter/lists_mergeby.py):

> allow the user to do `[list1, list2, ...] | lists_mergeby('index')`

This option is not documented. Also, the examples are missing.

### Issue Type

Bug Report

### Component Name

community.general.lists_mergeby

### Ansible Version

```ini
ansible [core 2.16.3]
  config file = /home/vlado/.ansible.cfg
  configured module search path = ['/home/vlado/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  ansible collection location = /home/vlado/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/vlado/.local/bin/ansible
  python version = 3.12.3 (main, Apr 10 2024, 05:33:47) [GCC 13.2.0] (/usr/bin/python3)
  jinja version = 3.1.2
  libyaml = True
```

### Community.general Version

```
community.general 9.0.1
```

### Configuration

```ini
CONFIG_FILE() = /export/home/vlado/tmp/ansible.cfg
DEFAULT_FILTER_PLUGIN_PATH(/export/home/vlado/tmp/ansible.cfg) = ['/home/vlado/tmp/plugins/filter']
DEFAULT_GATHERING(/export/home/vlado/tmp/ansible.cfg) = explicit
DEFAULT_HOST_LIST(/export/home/vlado/tmp/ansible.cfg) = ['/home/vlado/tmp/hosts']
DEFAULT_LOG_PATH(/export/home/vlado/tmp/ansible.cfg) = /var/log/ansible.log
DEFAULT_STDOUT_CALLBACK(/export/home/vlado/tmp/ansible.cfg) = yaml
```

### OS / Environment

```ini
shell> cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=24.04
DISTRIB_CODENAME=noble
DISTRIB_DESCRIPTION="Ubuntu 24.04 LTS"
```

### Steps to Reproduce

See: 

https://docs.ansible.com/ansible/latest/collections/community/general/lists_mergeby_filter.html

### Expected Results

NA

### Actual Results

NA

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by vbotka**
##### SUMMARY

Fix #8474

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

community.general.lists_mergeby

##### ADDITIONAL INFORMATION

NA

### CODE REVIEW

File: plugins/filter/lists_mergeby.py [hunk 1]
```diff
@@ -12,22 +12,32 @@
...
-      is used. To learn details on how to use the parameters O(recursive) and O(list_merge) see
-      Ansible User's Guide chapter "Using filters to manipulate data" section "Combining
-      hashes/dictionaries".
+    - Merge two or more lists by attribute O(index). Optional
+      parameters O(recursive) and O(list_merge) control the merging of
+      the nested dictionaries and lists.
+    - The function C(merge_hash) from C(ansible.utils.vars) is used.
+    - To learn details on how to use the parameters O(recursive) and
+      O(list_merge) see Ansible User's Guide chapter "Using filters to
+      manipulate data" section B(Combining hashes/dictionaries) or the
```
Comment (felixfontein) [inline plugins/filter/lists_mergeby.py:21]:
```suggestion
      manipulate data" section R(Combining hashes/dictionaries, combine_filter) or the
```

File: plugins/filter/lists_mergeby.py [hunk 2]
```diff
@@ -12,22 +12,32 @@
...
+      filter C(ansible.builtin.combine).
```
Comment (felixfontein) [inline plugins/filter/lists_mergeby.py:22]:
```suggestion
      filter P(ansible.builtin.combine#filter).
```

File: plugins/filter/lists_mergeby.py [hunk 3]
```diff
@@ -12,22 +12,32 @@
...
+
   positional: another_list, index
   options:
     _input:
-      description: A list of dictionaries.
+      description:
+        - A list of dictionaries or a list of lists of dictionaries.
```
Comment (felixfontein) [inline plugins/filter/lists_mergeby.py:28]:
```suggestion
        - A list of dictionaries, or a list of lists of dictionaries.
```

File: plugins/filter/lists_mergeby.py [hunk 4]
```diff
@@ -12,22 +12,32 @@
...
+          because all elements of O(_input) can be either dictionaries
+          or lists.
       type: list
-      elements: dictionary
+      elements: raw
       required: true
     another_list:
-      description: Another list of dictionaries. This parameter can be specified multiple times.
+      description:
+        - Another list of dictionaries or a list of lists of dictionaries.
```
Comment (felixfontein) [inline plugins/filter/lists_mergeby.py:37]:
```suggestion
        - Another list of dictionaries, or a list of lists of dictionaries.
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contriution!

**Comment by vbotka** [comment]
Done. Thank you for the review!

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@vbotka thanks for improving the docs!

**Comment by vbotka** [comment]
@felixfontein thank you!

**Pull request merged**